cypress/tests/ui/flowchart/flowchart.cy.js

```diff
@@ -1 +1,166 @@
 // All E2E Tests Related to flowchart goes here.
+describe('Flowchart', () => {
+  it('verifies that users can expand a collapsed modular pipeline in the flowchart. #TC-23', () => {
+    const modularPipelineText = 'feature_engineering';
+    const taskNodeText = 'Create Derived Features';
+
+    // Assert before action
+    cy.get('.pipeline-node > .pipeline-node__text').should(
+      'not.contain',
+      taskNodeText
+    );
+
+    // Action
+    cy.get(`.pipeline-node[data-id=${modularPipelineText}]`).click();
+
+    // Assert after action
+    cy.get('.pipeline-node > .pipeline-node__text').should(
+      'contain',
+      taskNodeText
+    );
+  });
+
+  it('verifies that users can open the metadata panel for nodes. #TC-24', () => {
+    // Assert before action
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+
+    // Action
+    cy.get('.pipeline-node.pipeline-node--task').first().click();
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+  });
+
+  it('verifies that users can open the code block in the metadata panel. #TC-25', () => {
+    // Assert before action
+    cy.get('.pipeline-metadata-code--visible').should('not.exist');
+
+    // Action
+    cy.get('.pipeline-node.pipeline-node--task').first().click();
+    cy.get('[data-test="pipeline-toggle-input-code"]').check({ force: true });
+
+    // Assert after action
+    cy.get('.pipeline-metadata-code--visible').should('exist');
+    cy.get('.pipeline-metadata-code__title').should('have.text', 'Code block');
+  });
+
+  it('verifies that users can open the metadata panel for parameters. #TC-26', () => {
+    const nodeToToggleText = 'Parameters';
+
+    // Alias
+    cy.get(`.pipeline-nodelist__row__checkbox[name=${nodeToToggleText}]`).as(
+      'nodeToToggle'
+    );
+
+    // Assert before action
+    cy.get('@nodeToToggle').should('not.be.checked');
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+
+    // Action
+    cy.get('@nodeToToggle').check({ force: true });
+    cy.get('.pipeline-node--parameters').first().click({ force: true });
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+  });
+
+  it('verifies that users can open the metadata panel for datasets. #TC-27', () => {
```

Comment:
nit - since we have a test that you click on node and metadata panel opens. we can remove this. also since we have the below test which checks for preview datasets